src/mem/ruby/protocol/GPU_VIPER-TCC.sm

```diff
@@ -837,17 +873,8 @@ machine(MachineType:TCC, "TCC Cache")
 
   transition(V, Atomic, M) {TagArrayRead, DataArrayWrite} {
     p_profileHit;
-    wdb_writeDirtyBytes;
-    pa_performAtomic;
-    ar_sendAtomicResponse;
-    p_popRequestQueue;
-  }
-
-  transition(I, Atomic, M) {TagArrayRead, TagArrayWrite, DataArrayWrite} {
-    p_profileMiss;
-    a_allocateBlock;
     ut_updateTag;
-    wdb_writeDirtyBytes;
+    owm_orWriteMask;
     pa_performAtomic;
     ar_sendAtomicResponse;
```

Comment:
This will send the modified data *after* the atomic, wouldn't it?  Atomics must return the previous data, before an atomic was performed.   I am still thinking we need to save the old data somewhere.  This applies for all transitions that call the ar_sendAtomicResponse action

Reply:
The previous data is stored in the atomic log, and read back from the atomic log in the coalescer per [this change](https://github.com/gem5/gem5/pull/200/files#diff-9e61b05abc867f6a980b3e43684b5e477ebd2b896a7e79aba0c573d5fa9a964bR591) in #200 .

Reply:
Ah, right.